Rollup of 14 pull requests

kennytm opened the pull request:
Successful merges:

 - #51619 (rust: add initial changes to support powerpc64le musl)
 - #51793 (Fix variant background color on hover in search results)
 - #52005 (Update LLVM to bring in a wasm codegen fix)
 - #52016 (Deduplicate error reports for statics)
 - #52019 ([cross-lang-lto] Allow the linker to choose the LTO-plugin (which is useful when using LLD))
 - #52030 (Any docs preposition change)
 - #52031 (Strenghten synchronization in `Arc::is_unique`)
 - #52033 ([Gardening] Update outdated comments: ByVal -> Scalar)
 - #52055 (Include VS 2017 in error message.)
 - #52063 (Add a link to the rustc docs)
 - #52073 (Add a punch card to weird expressions test)
 - #52080 (Improve dependency deduplication diagnostics)
 - #52093 (rustc: Update tracking issue for wasm_import_module)
 - #52096 (Fix typo in cell.rs)

Failed merges:

kennytm commented:
@bors r+ p=13

bors commented:
:pushpin: Commit 0a3756b5b0b1aff866972e171e9e6d367de2301c has been approved by `kennytm`

<!-- @bors r=kennytm 0a3756b5b0b1aff866972e171e9e6d367de2301c -->

bors commented:
:hourglass: Testing commit 0a3756b5b0b1aff866972e171e9e6d367de2301c with merge 67b3b2d62abb8f7360050be64d9ba81ebcc6c2fb...

kennytm commented:
@bors r- retry Failed CI

kennytm commented:
@bors r+

Removed #51953 (failed CI)
Removed #51861 (already testing)
Added #52055 
Added #52063

bors commented:
:pushpin: Commit 22b01e4a8230c2a6a97c3f94925c0bfcf130e6ef has been approved by `kennytm`

<!-- @bors r=kennytm 22b01e4a8230c2a6a97c3f94925c0bfcf130e6ef -->

bors commented:
:pushpin: Commit 5da9fdc19831f3f55592f4fb94db303213b55d81 has been approved by `kennytm`

<!-- @bors r=kennytm 5da9fdc19831f3f55592f4fb94db303213b55d81 -->

bors commented:
:hourglass: Testing commit 5da9fdc19831f3f55592f4fb94db303213b55d81 with merge bff009016b8a91485a4be7fee9aad946d59f47f0...

bors commented:
:broken_heart: Test failed - [status-appveyor](https://ci.appveyor.com/project/rust-lang/rust/build/1.0.8044)

kennytm commented:
@bors retry 

A build script segfault, assuming spurious for now.

```
error: failed to run custom build command for `unicase v1.4.0`
process didn't exit successfully: `C:\projects\rust\build\ct\iron\target\debug\build\unicase-38624802fb981d7a\build-script-build` (exit code: 101)
--- stderr
thread 'main' panicked at 'unexpected -vV format', C:\Users\appveyor\.cargo\registry\src\github.com-1ecc6299db9ec823\rustc_version-0.1.7\src\lib.rs:126:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

bors commented:
:hourglass: Testing commit 5da9fdc19831f3f55592f4fb94db303213b55d81 with merge 01c430a5043cc55a2f2ac4e9c8d6f8384cefd0a1...

bors commented:
:broken_heart: Test failed - [status-appveyor](https://ci.appveyor.com/project/rust-lang/rust/build/1.0.8046)

kennytm commented:
Heh, okay.

@bors r+

Removed #52052 (see comment above)
Removed #51953 (already testing)
Added back #52093 (should fixed test)
Added #52096

bors commented:
:pushpin: Commit e6ddbabb5998d1a83918831f3859a95a2adb110c has been approved by `kennytm`

<!-- @bors r=kennytm e6ddbabb5998d1a83918831f3859a95a2adb110c -->

bors commented:
:hourglass: Testing commit e6ddbabb5998d1a83918831f3859a95a2adb110c with merge 4d9fa2326e184314749ccf79203f5ecc35e6225c...

bors commented:
:sunny: Test successful - [status-appveyor](https://ci.appveyor.com/project/rust-lang/rust/build/1.0.8048), [status-travis](https://travis-ci.org/rust-lang/rust/builds/400755586?utm_source=github_status&utm_medium=notification)
Approved by: kennytm
Pushing 4d9fa2326e184314749ccf79203f5ecc35e6225c to master...
